hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/SCMPipelineManager.java

```diff
@@ -162,6 +183,16 @@ public synchronized Pipeline createPipeline(ReplicationType type,
       if (pipeline.isOpen()) {
         metrics.incNumPipelineCreated();
         metrics.createPerPipelineMetrics(pipeline);
+        if (RatisPipelineUtils.checkPipelineContainSameDatanodes(
```

Comment:
Returned previous pipeline is not recorded.

Reply:
There are another case of pipeline open, through openPipeline API. You should add the same logic there.

Reply:
Updated.